### PR TITLE
Feature/bounding box check

### PR DESCRIPTION
**Pull request by rikba**
Check polynomials if they are inside a set of half planes.

The check projects the polynomial into half plane direction, computes the roots of the derivative and evaluate the critical points.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -310,14 +310,14 @@ presented in [Müller2015](http://flyingmachinearena.org/wp-content/publications
   publisher={IEEE}
 }
 ```
-We adapted [https://github.com/markwmuller/RapidQuadrocopterTrajectories] to
+We adapted https://github.com/markwmuller/RapidQuadrocopterTrajectories to
```
Comment (helenol) [inline README.md]: Why undo the link? [RapidQuadrocopterTrajectories](https://github.com/markwmuller/RapidQuadrocopterTrajectories) is my recommendation
```
[RapidQuadrocopterTrajectories](https://github.com/markwmuller/RapidQuadrocopterTrajectories) 
```

File: README.md [hunk 2]
```diff
@@ -352,3 +352,39 @@ this.
 Here are the computational results over 1000 random segments with different
 parameter settings:
 <p align="center"><img src="https://cloud.githubusercontent.com/assets/11293852/26199226/903dea9a-3bc9-11e7-945e-91e5a119e63f.png" /></p>
+
+## Checking Half-Space Feasibility
+The package also contains a method to check if a trajectory or segment is inside
+an arbitrary set of half spaces based on https://github.com/markwmuller/RapidQuadrocopterTrajectories.
```
Comment (helenol) [inline README.md]: also link here

File: mav_trajectory_generation_ros/test/test_feasibility.cpp [hunk 3]
```diff
@@ -286,6 +292,57 @@ TEST(PolynomialTest, CompareFeasibilityTests) {
   }
 }
 
+TEST(PolynomialTest, HalfPlaneFeasibility) {
```
Comment (helenol) [inline mav_trajectory_generation_ros/test/test_feasibility.cpp]: Also change to FeasibilityTest here?

File: mav_trajectory_generation_ros/test/test_feasibility.cpp [hunk 4]
```diff
@@ -286,6 +292,57 @@ TEST(PolynomialTest, CompareFeasibilityTests) {
...
+
+  FeasibilityBase box_check;
+  Eigen::Vector3d box_center(0.0, 0.0, 0.0);
+
+  // Grow box.
+  double l = 0.0;
+  while(l < 4.0) {
+    Eigen::Vector3d box_size(Eigen::Vector3d::Constant(l));
+    box_check.half_plane_constraints_ = HalfPlane::createBoundingBox(box_center, box_size);
+    for (const HalfPlane& half_plane : box_check.half_plane_constraints_) {
```
Comment (helenol) [inline mav_trajectory_generation_ros/test/test_feasibility.cpp]: Empty for loop?

Comment (rikba) [inline mav_trajectory_generation_ros/test/test_feasibility.cpp]: :dagger:

**Pull request merged**